service/frontend/workflowHandler.go

```diff
@@ -4040,6 +4040,8 @@ func checkRequiredDomainDataKVs(requiredDomainDataKeys map[string]interface{}, d
 func (wh *WorkflowHandler) normalizeVersionedErrors(ctx context.Context, err error) error {
 	switch err.(type) {
 	case *types.WorkflowExecutionAlreadyCompletedError:
+		return &types.EntityNotExistsError{Message: "Workflow execution already completed."}
+		/* TODO: re-enable the block below once we can rollout this *breaking* change
```

Comment:
might be worth adding a note to `versionChecker.SupportsWorkflowAlreadyCompletedError` too, but 👍 this makes it nice and clear for this temporary period.